lib/ui/src/core/store.js

```diff
@@ -29,11 +32,11 @@ export default class Store {
 
   // The assumption is that this will be called once, to initialize the React state
   // when the module is instanciated
-  getInitialState() {
+  getInitialState(base = {}) {
     // We don't only merge at the very top level (the same way as React setState)
     // when you set keys, so it makes sense to do the same in combining the two storage modes
     // Really, you shouldn't store the same key in both places
-    return { ...get(localStorage), ...get(sessionStorage) };
+    return merge(base, { ...get(localStorage), ...get(sessionStorage) });
```

Comment:
Again, I think this make break things. The idea was that modules would pick the relevant keys out of the persisted state and combine sensibly with their defaults, rather than us doing anything automatically via `merge`.

There are a bunch of ways this can break (doing it automatically).

Reply:
reverted